packages/shared/client/types.ts

```diff
@@ -215,6 +211,14 @@ export interface ReplayClientInterface {
     },
     onMatches: (matches: SearchSourceContentsMatch[], didOverflow: boolean) => void
   ): Promise<void>;
+  streamAnalysis(
+    params: AnalysisParams,
+    handlers: {
+      onPoints?: (points: PointDescription[]) => void;
+      onResults?: (results: AnalysisEntry[]) => void;
+      onError?: (error: any) => void;
+    }
+  ): { pointsReceived: Promise<void>; resultsReceived: Promise<void> };
```

Comment:
nit: Would "finished" be clearer than "received"? Seems like "received" could potentially mean either "first result received" or "all results received"